smart-frontend/app/src/components/pages/RulesPage/RulesDialogs/RuleCreateDialog.tsx

```diff
@@ -15,45 +15,39 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import React, { useEffect, useState } from 'react';
+import React, { useCallback, useEffect, useRef } from 'react';
 import { useDispatch, useStore } from '@hooks';
 import { closeCreateRuleDialog, createRuleWithUpdate } from '@store/adh/rules/rulesActionsSlice';
-import { FooterDialog, MultilineInput } from '@uikit';
+import { FooterDialog } from '@uikit';
 import { SpinnerPanel } from '@uikit/Spinner/Spinner';
+import MonacoCodeEditor from '@uikit/MonacoCodeEditor/MonacoCodeEditor';
 
 const RuleCreateDialog: React.FC = () => {
   const dispatch = useDispatch();
   const isOpen = useStore(({ adh }) => adh.rulesActions.createDialog.isOpen);
   const isActionInProgress = useStore(({ adh }) => adh.rulesActions.isActionInProgress);
 
-  const [ruleText, setRuleText] = useState('');
+  const ruleText = useRef('');
 
   useEffect(() => {
     // clear when close dialog
     if (!isOpen) {
-      setRuleText('');
+      ruleText.current = '';
     }
-  }, [isOpen, setRuleText]);
+  }, [isOpen]);
 
   const closeDialog = () => {
     dispatch(closeCreateRuleDialog());
   };
 
-  const handleChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
-    setRuleText(event.target.value);
-  };
+  const handleChange = useCallback((value: string) => {
+    ruleText.current = value;
+  }, []);
 
   const handleCreate = () => {
     dispatch(createRuleWithUpdate(ruleText));
   };
 
-  const handleKeyDown = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
```

Comment:
can we try handle hot key for fast save?

Reply:
fixedm, thank you